lib/retriable.rb

```diff
@@ -55,14 +55,10 @@ def retriable(opts = {})
         end
       rescue *[*exception_list] => exception
         if on.kind_of?(Hash)
-          patterns = [*on[exception.class]]
-          message_match = patterns.empty? ? true : false
-          patterns.each do |pattern|
-            if !!(exception.message =~ pattern)
-              message_match = true
-              break
-            end
+          message_match = exception_list.select { |e| exception.is_a?(e) }.inject(false) do |match, e|
+            match || [*on[e]].empty? || [*on[e]].any? { |pattern| exception.message =~ pattern }
```

Comment:
Don't we only care about the first match? If the first iteration matches, we can stop looking right? The reduce here means even if you match the first exception, it'll keep looking even though it doesn't have to.